test/jobs/update_developer_response_rate_job_test.rb

```diff
@@ -0,0 +1,81 @@
+require "test_helper"
+
+class UpdateDeveloperResponseRateJobTest < ActiveJob::TestCase
+  include BusinessesHelper
+  setup do
+    @developer = developers(:one)
+
+    # Create a conversation that has been replied to.
+    business = Business.create!(business_attributes)
+    conversation = Conversation.create!(
+      developer: @developer,
+      business: business,
+      created_at: UpdateDeveloperResponseRateJob::GRACE_PERIOD.ago - 1.day
+    )
+    Message.create!(conversation: conversation, sender: business, body: "First contact")
+    Message.create!(conversation: conversation, sender: @developer, body: "First reply")
+
+    # Create a conversation that has not been replied to.
+    business = Business.create!(business_attributes)
+    conversation = Conversation.create!(
+      developer: @developer,
+      business: business,
+      created_at: UpdateDeveloperResponseRateJob::GRACE_PERIOD.ago - 1.day
+    )
+    Message.create!(conversation: conversation, sender: business, body: "Second contact")
+
+    # update the developer's response rate
+    UpdateDeveloperResponseRateJob.perform_now(@developer.reload)
+  end
+
+  test "should update developer response rate from the returned value" do
+    replied_rate = 0.42
+    expected_response_rate = 42
+    mock = Minitest::Mock.new
+    mock.expect :replied_rate, replied_rate
```

Comment:
Why a song and dance with the mock instead of checking for `50`? We clearly create both a replied to and not replied to conversation in the setup of this test, so no risk for mystery guests.

Reply:
This is my brain trying to translate RSpec to MiniTest. 🙃
I'll see if I can make it clearer.

Reply:
I have done an almost complete rewrite of the `test/jobs/update_developer_response_rate_job_test.rb` 
I've learn't a lot. I see more MiniTest in my future! :crystal_ball: